### ISSUE TITLE
awesome plugin!! i can help you simplify it!

### ISSUE THREAD

**Issue by acao** [cimdalli/graphiql-plugin-chatgpt#1]
this is a really cool usage of our plugin api!

we now have made it so that query and setQuery are internal details of a plugin. you can access current query and setQuery callback internally! see the changelog for graphiql explorer plugin. this prevents cursor jumping bugs and extra re-rendering. setting query and handler of the GraphiQL component leads to a semi controlled state that is not always ideal.

**Comment by acao** [comment]
@cimdalli here is the preferred pattern:

https://github.com/graphql/graphiql/blob/main/packages/graphiql-plugin-explorer/CHANGELOG.md#minor-changes

your implementation no longer needs to require the user creating a volatile `useState()` implementation

**Comment by cimdalli** [comment]
@acao This is awesome, we have jumping cursor issue with the explorer plugin but seems it's solved already. I will apply your suggestion asap.

**Comment by acao** [comment]
@cimdalli indeed, the solution is to avoid passing volatile setState to both the plugin and GraphiQL when it can be avoided, the reccomendation is for plugins to [retrieve and set state internally using hooks](https://github.com/graphql/graphiql/blob/9049e79ff7531f34e4d2ddbcbc7fed7a444760fc/packages/graphiql-plugin-explorer/src/index.tsx#L142) as you can see now in the explorer plugin source for example. 

for controlled component cases we will need a better solution, as I believe the cursor jump will still happen there
